src/shogun/base/SGObject.h

```diff
@@ -390,9 +393,34 @@ class CSGObject
 		put(Tag<T*>(name), value);
 	}
 
+    /** Typed setter for an enum mapped with a string, identified by a name.
+     *
+     * @param name name of the parameter
+     * @param value value of the parameter
+	 */
+	template <class T>
+	void put(const std::string& name, const std::string& value) noexcept(false)
+	{
+		if (m_enum_to_string_map.find(get_name()) == m_enum_to_string_map.end())
+			SG_ERROR("There is no enum mapping for the %s", get_name());
+
+		auto string_to_enum_map = m_enum_to_string_map[get_name()];
+
+		if (string_to_enum_map.find(name) == string_to_enum_map.end())
+			SG_ERROR("There is no enum mapping for the parameter %s::%s", get_name(), name);
+
+		auto string_to_enum = string_to_enum_map[name];
+
+		if (string_to_enum.find(value) == string_to_enum.end())
+			SG_ERROR("There is no enum mapping for '%s' in %s::%s", value, get_name(), name);
```

Comment:
Thinking this further: I wouldnt even talk about enum mappings, but just “options”. Users don’t need to know that there is an underlying enum